packages/react/src/components/Geo/LocationSearch/index.tsx

```diff
@@ -15,8 +20,35 @@ const LOCATION_SEARCH_OPTIONS = {
 
 const LOCATION_SEARCH_CONTAINER = 'geocoder-container';
 
+interface LocationSearchProps extends MaplibreGeocoderOptions {
+  /**
+   * Emitted when the input is cleared
+   */
+  onClear: () => void;
+  /**
+   * Emitted when the geocoder is looking up a query
+   */
+  onLoading: (query) => void;
+  /**
+   * Fired when the geocoder returns a response
+   */
+  onResults: (results: MaplibreGeocoderResults) => void;
+  /**
+   * Fired when input is set
+   */
+  onResult: (result: MaplibreGeocoderResult) => void;
+  /**
+   * Emitted on error as string
+   */
+  onError: (error) => void;
+}
```

Comment:
Make these methods optional as they are not required on all usages of LocationSearch.